src/stk/itk_bridge/itk_convert.h

```diff
@@ -0,0 +1,159 @@
+
+    // Copy metadata
+    const auto spacing = image->GetSpacing();
+    volume.set_spacing({float(spacing[0]), float(spacing[1]), float(spacing[2])});
+
+    const auto origin = image->GetOrigin();
+    volume.set_origin({float(origin[0]), float(origin[1]), float(origin[2])});
+
+    const auto direction = image->GetDirection();
+    matrix<float, 3, 3> stk_direction;
+    for (unsigned int i = 0; i < direction.RowDimensions; ++i) {
+        for (unsigned int j = 0; j < direction.ColumnDimensions; ++j) {
+            stk_direction(i, j) = direction(i, j);
+        }
+    }
+    volume.set_direction(stk_direction);
+
+    // Store other metadata
+    auto const& metadata = image->GetMetaDataDictionary();
+    for (auto const& k : metadata.GetKeys()) {
+        std::string v;
+        itk::ExposeMetaData<std::string>(metadata, k, v);
+        volume.set_metadata(k, v);
+    }
+
+    return volume;
+}
+
+
+/*!
+ * \brief Convert an STK object into an ITK object.
+ * \note The new object owns a deep copy of the image data.
+ */
+template<class ImageType>
+void stk2itk(const stk::Volume& volume, typename ImageType::Pointer& image)
+{
+    using IndexType = typename ImageType::IndexType;
+    using SizeType = typename ImageType::SizeType;
+    using SizeValueType = typename ImageType::SizeValueType;
+    using RegionType = typename ImageType::RegionType;
+    using PointType = typename ImageType::PointType;
+    using PointValueType = typename ImageType::PointValueType;
+    using SpacingType = typename ImageType::SpacingType;
+    using SpacingValueType = typename ImageType::SpacingValueType;
+    using DirectionType = typename ImageType::DirectionType;
+    using PixelType = typename ImageType::PixelType;
+
+    const IndexType start = {0, 0, 0};
+    const SizeType size = {
+        SizeValueType(volume.size().x),
+        SizeValueType(volume.size().y),
+        SizeValueType(volume.size().z),
+    };
+    PointType origin;
+    origin[0] = PointValueType(volume.origin().x);
+    origin[1] = PointValueType(volume.origin().y);
+    origin[2] = PointValueType(volume.origin().z);
+    SpacingType spacing;
+    spacing[0] = SpacingValueType(volume.spacing().x);
+    spacing[1] = SpacingValueType(volume.spacing().y);
+    spacing[2] = SpacingValueType(volume.spacing().z);
+    DirectionType direction;
+    auto const stk_direction = volume.direction();
+    for (unsigned int i = 0; i < stk_direction.rows; ++i) {
+        for (unsigned int j = 0; j < stk_direction.cols; ++j) {
+            direction(i, j) = stk_direction(i, j);
+        }
+    }
+
+    RegionType region;
+    region.SetIndex(start);
+    region.SetSize(size);
+
+    image = ImageType::New();
+    image->SetRegions(region);
+    image->SetOrigin(origin);
+    image->SetSpacing(spacing);
+    image->SetDirection(direction);
+    image->Allocate();
+
+    const size_t voxel_no = size[0] * size[1] * size[2];
+
+    // Copy image data
+    if (volume.is_contiguous()) {
```

Comment:
I added handling for non-contiguous volumes. Hoping I was not hallucinating too much when I wrote this...

Reply:
stk side of things seems correct.